NOAA-OWP/cahaba
# [3pt] PR: Enhancements to allow for evaluation at AHPS sites and generation of categorical FIM

Enhancements to allow for evaluation at AHPS sites and generation of categorical FIM. Evaluations at AHPS sites (NWS and USGS) are now possible.

This enhancement relies on a new directory structure in the `test_cases` directory. After this is merged into `dev` it will be necessary to update the `test_cases` directory with the new structure. `synthesize_test_cases.py` also produces a "master" CSV of metrics from all versions that allows for easy queries and plotting.

This PR addresses #258 and #259.

## Additions

- `generate_categorical_fim.py`

## Removals

- `ble_autoeval.sh`
- `all_ble_stats_comparison.py`
- The code block that is responsible for printing the colored metrics to screen has now been commented out because of the new scale of evaluations (formerly in `run_test_case.py`, now in `shared_functions.py`)

## Changes

- `synthesize_test_cases.py`
- `run_test_case.py`
- `shared_functions.py`
- `shared_variables.py`
- `cache_metrics.py`

## Testing

1. Run `synthesize_test_cases.py` in DEV and PREV mode (-c)
2. Run `generate_categorical_fim.py`

Please confirm that these scripts run without error. The help (-h) ought to be fairly good for these.
